### PR TITLE
Documentation improvements for stairs()

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Addresses bullet points 1 and 3 of https://github.com/matplotlib/matplotlib/pull/18275#pullrequestreview-487347047:

- Comparison of `.pyplot.step` and `.pyplot.stairs` could use some explanatory text.
- Some docstring parameters don't have a description. Even if they state something apparently obvious, they should be explicit.

### CODE REVIEW

File: doc/users/next_whats_new/steppatch_and_stairs.rst [hunk 1]
```diff
@@ -1,7 +1,11 @@
 New StepPatch artist and a stairs method
 ----------------------------------------
-New `~.matplotlib.patches.StepPatch` artist and  `.pyplot.stairs` method.
-For both the artist and the function, the x-like edges input is one 
+`.pyplot.stairs` and the underlying artist `~.matplotlib.patches.StepPatch`
+provide a cleaner interface for plotting stepwise constant functions for the
+common case that you know the step edges. This superseeds most use cases of
+`.pyplot.step`.
```
Comment (jklymak) [inline doc/users/next_whats_new/steppatch_and_stairs.rst:6]:
```suggestion
common case that you know the step edges. This superseeds many use cases of
`.pyplot.step`, for instance when plotting the output of `numpy.histogram`.
```

File: doc/users/next_whats_new/steppatch_and_stairs.rst [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+
+For both the artist and the function, the x-like edges input is one
```
Comment (jklymak) [inline doc/users/next_whats_new/steppatch_and_stairs.rst:8]:
```suggestion
For both the artist and the function, the x-like edges input is one element
```

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -2036,6 +2036,15 @@ def step(self, x, y, *args, where='pre', data=None, **kwargs):
         formatting options. Most of the concepts and parameters of plot can be
         used here as well.
 
+        .. note::
+
+            This method uses a standard plot with a step drawstyle: The *x*
+            values are the reference positions and steps extend left/right/both
+            directions depending on *where*.
+
+            For the common case, that you know the values and edges of the
```
Comment (jklymak) [inline lib/matplotlib/axes/_axes.py:2045]:
```suggestion
            For the common case where you know the values and edges of the
```

File: examples/lines_bars_and_markers/stairs_demo.py [hunk 4]
```diff
@@ -4,26 +4,23 @@
 ===========
 
 This example demonstrates the use of `~.matplotlib.pyplot.stairs`
-for histogram and histogram-like data visualization and an associated
-underlying `.StepPatch` artist, which is a specialized version of
-`.PathPatch` specified by its bins and edges.
+for piecewise constant functions. A common use case is histogram and
```
Comment (jklymak) [inline examples/lines_bars_and_markers/stairs_demo.py:7]:
```suggestion
for stepwise constant functions. A common use case is histogram and
```
You used "stepwise" below, which is less precise, but I think more colloquial.

**Comment by jklymak** [comment]
pinging @andrzejnovak, in case they see any issues.  But this looks good to me...

**Comment by andrzejnovak** [comment]
Thanks, LGTM.

**Review by jklymak — APPROVED** [review]
Anyone can merge once CI is happy....

**Pull request merged**